dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/enrollment/AbstractEnrollmentService.java

```diff
@@ -372,9 +373,14 @@ public Enrollment getEnrollment( User user, ProgramInstance programInstance, Tra
         {
             for ( RelationshipItem relationshipItem : programInstance.getRelationshipItems() )
             {
-                enrollment.getRelationships()
-                    .add( relationshipService.getRelationship( relationshipItem.getRelationship(),
-                        RelationshipParams.FALSE, user ) );
+                org.hisp.dhis.relationship.Relationship daoRelationship = relationshipItem.getRelationship();
+                if ( trackerAccessManager.canRead( user, daoRelationship ).isEmpty()
```

Comment:
what does `canRead().isEmpty()` mean?
should this not just be `canRead()`, so if the user is allowed to read the rel we fetch more details on it

Reply:
I agree this api is confusing, apparently canRead returns a list of errors and if there is none then it is ok to read.
I am just using the API here, I am not going down this api, even more as it is part of the old importer